scripts/circle-env.js

```diff
@@ -20,8 +20,15 @@ async function checkCanaries () {
 
   const circleEnv = await readCircleEnv()
 
-  if (Object.keys(circleEnv).length === 0) {
-    return console.warn('CircleCI env empty, assuming this is a contributor PR. Not checking for canary variables.')
+  // if the config contains only CIRCLE_OIDC_TOKEN, CIRCLE_OIDC_TOKEN_V2, CIRCLE_PLUGIN_TEST, treat the config as if it were empty
+  const containsOnlyAllowedEnvs = () => {
+    const circleEnvKeys = Object.keys(circleEnv)
+
+    return circleEnvKeys.length === 3 && circleEnvKeys.includes('CIRCLE_OIDC_TOKEN') && circleEnvKeys.includes('CIRCLE_OIDC_TOKEN_V2') && circleEnvKeys.includes('CIRCLE_PLUGIN_TEST')
+  }
+
+  if (Object.keys(circleEnv).length === 0 || containsOnlyAllowedEnvs()) {
+    return console.warn('CircleCI env empty or contains only allowed envs, assuming this is a contributor PR. Not checking for canary variables.')
```

Comment:
```suggestion
  // if the config contains only CIRCLE_OIDC_TOKEN, CIRCLE_OIDC_TOKEN_V2, CIRCLE_PLUGIN_TEST, treat the config as if it were empty
  const containsOnlyAllowedEnvs = () => {
    const circleEnvKeys = Object.keys(circleEnv)

    return Object.keys(circleEnv).length === 0 || (circleEnvKeys.length === 3 && circleEnvKeys.includes('CIRCLE_OIDC_TOKEN') && circleEnvKeys.includes('CIRCLE_OIDC_TOKEN_V2') && circleEnvKeys.includes('CIRCLE_PLUGIN_TEST'))
  }

  if (containsOnlyAllowedEnvs()) {
    return console.warn('CircleCI env empty or contains only allowed envs, assuming this is a contributor PR. Not checking for canary variables.')
```

Reply:
i think that works? blend the length zero check into contains...

Reply:
rebased in 8050b2e5d4